ui/src/app/pages/workspace/workspace-edit.spec.tsx

```diff
@@ -411,13 +411,72 @@ describe('WorkspaceEdit', () => {
     expect(mockNavigate).toHaveBeenCalledTimes(1);
   });
 
-  // TODO RW-7973
-  // it('supports successful duplication in asynchronous mode', async () => {
-  //   workspaceEditMode = WorkspaceEditMode.Duplicate;
-  //   environment.enableAsyncWorkspaceOperations = true;
-  // });
+  it('supports successful duplication in asynchronous mode', async () => {
+    environment.enableAsyncWorkspaceOperations = true;
+    workspaceEditMode = WorkspaceEditMode.Duplicate;
+    const wrapper = component();
+    await waitOneTickAndUpdate(wrapper);
+
+    wrapper
+      .find('[data-test-id="review-request-btn-false"]')
+      .first()
+      .simulate('click');
+    await waitOneTickAndUpdate(wrapper);
+
+    const numBefore = workspacesApi.workspaceOperations.length;
+    wrapper
+      .find('[data-test-id="workspace-save-btn"]')
+      .first()
+      .simulate('click');
+    await waitOneTickAndUpdate(wrapper);
+
+    wrapper
+      .find('[data-test-id="workspace-confirm-save-btn"]')
+      .first()
+      .simulate('click');
+    await waitOneTickAndUpdate(wrapper);
+    expect(workspacesApi.workspaceOperations.length).toEqual(numBefore + 1);
+    expect(mockNavigate).toHaveBeenCalledTimes(1);
+  });
+
+  it('defaults to upgrading the CDR Version when synchronously duplicating a workspace with an older CDR Version', async () => {
```

Comment:
Is this test and next one are essential same but with different value in flag `environment.enableAsyncWorkspaceOperations`? If so, I recommend using `it.each`.

Reply:
not completely - one counts `workspaces` and the other `workspaceOperations`